src/workerd/server/workerd-api.c++

```diff
@@ -629,26 +629,28 @@ static v8::Local<v8::Value> createBindingValue(
 void WorkerdApiIsolate::compileGlobals(
     jsg::Lock& lockParam, kj::ArrayPtr<const Global> globals,
     v8::Local<v8::Object> target,
+    kj::Maybe<v8::Local<v8::Object>> processEnv,
     uint32_t ownerId) const {
   auto& lock = kj::downcast<JsgWorkerdIsolate::Lock>(lockParam);
-  v8::HandleScope scope(lock.v8Isolate);
+  auto isolate = lock.v8Isolate;
+  v8::HandleScope scope(isolate);
   auto context = lock.v8Context();
   auto& featureFlags = *impl->features;
 
   for (auto& global: globals) {
-    v8::HandleScope scope(lock.v8Isolate);
-
     // Don't use String's usual TypeHandler here because we want to intern the string.
-    auto name = jsg::v8StrIntern(lock.v8Isolate, global.name);
+    auto name = jsg::v8StrIntern(isolate, global.name);
     auto value = createBindingValue(lock, global, featureFlags);
```

Comment:
Complete side note: I have to wonder if there's any value at all in lazily instantiating bindings in terms of reducing cold start times. Probably not significant enough to spend the time implementing but something we could probably investigate later?